src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -208,6 +208,102 @@ internal sealed override IEnumerable<Cci.INestedTypeDefinition> GetSynthesizedNe
             return null;
         }
 
+        public sealed override List<(Cci.ITypeDefinition, Cci.DebugSourceDocument[])> GetTypeToDebugDocumentMap(EmitContext context)
+        {
+            var debugDocuments = ArrayBuilder<Cci.DebugSourceDocument>.GetInstance();
+            var methodDocumentList = PooledHashSet<Cci.DebugSourceDocument>.GetInstance();
+            var result = new List<(Cci.ITypeDefinition, Cci.DebugSourceDocument[])>();
+
+            var namespacesAndTypesToProcess = new Stack<NamespaceOrTypeSymbol>();
```

Comment:
nit: we often use pooled `ArrayBuilder<T>` as a stack. For example: https://github.com/dotnet/roslyn/blob/2eddced21a8d9dcfea0509dec368be07644b4302/src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs#L2234

Reply:
It's API compatible <3